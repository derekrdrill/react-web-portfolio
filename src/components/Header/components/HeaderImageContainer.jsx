import React from 'react';
import { Link as ScrollLink } from 'react-scroll';
import PropTypes from 'prop-types';

export const HeaderImageContainer = ({ children, condition }) =>
  condition ? <ScrollLink to='home'>{children}</ScrollLink> : children;

HeaderImageContainer.propTypes = {
  children: PropTypes.node,
  condition: PropTypes.bool,
};
