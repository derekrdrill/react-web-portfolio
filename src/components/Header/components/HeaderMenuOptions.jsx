import React from 'react';
import PropTypes from 'prop-types';
import { Link as RouterLink } from 'react-router-dom';
import { Link as ScrollLink } from 'react-scroll';
import styled from 'styled-components';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

// TODO: revisit condensing the two types of links with css instead of styled components
// import { css } from '@emotion/react';

export const HeaderMenuOptions = ({
  menuTitle,
  menuIcon,
  mainTo,
  secondaryTo,
  headerType,
  menuType,
}) => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return headerType === 'main' ? (
    <StyledScrollLink
      activeClass='active'
      darkMode={darkMode}
      duration={500}
      menuType={menuType}
      offset={-70}
      spy={true}
      to={mainTo}
    >
      {menuTitle}
      {menuIcon}
    </StyledScrollLink>
  ) : (
    <StyledRouterLink darkMode={darkMode} menuType={menuType} to={secondaryTo}>
      {menuTitle}
      {menuIcon}
    </StyledRouterLink>
  );
};

HeaderMenuOptions.propTypes = {
  menuTitle: PropTypes.string,
  menuIcon: PropTypes.node,
  mainTo: PropTypes.string,
  secondaryTo: PropTypes.string,
  headerType: PropTypes.string,
  menuType: PropTypes.string,
};

// TODO: revisit condensing the two types of links with css instead of styled components

// const styledLink = () => css({
//     color: 'royalblue',
//     textDecoration: 'none',
//     cursor: 'pointer',
//     backgroundImage: 'linear-gradient(royalblue, royalblue)',
//     backgroundSize: '0% 0.1em',
//     backgroundPositionY: '100%',
//     backgroundPositionX: '100%',
//     backgroundRepeat: 'no-repeat',
//     margin: 15,
//     padding: 2,
//     transition: 'background-size 0.2s ease-in-out',
//     textTransform: 'uppercase',
//     svg: {
//         transform: 'translate(3px, -2px)',
//         width: 17,
//         height: 17,
//     },
//     ':hover': {
//         backgroundSize: '100% 0.1em',
//         backgroundPositionX: '0%',
//     },
// });

export const StyledScrollLink = styled(ScrollLink)(({ darkMode, menuType }) => ({
  backgroundImage:
    menuType === 'list'
      ? 'linear-gradient(#212121, #212121)'
      : !darkMode
      ? 'linear-gradient(darkslategray, darkslategray)'
      : 'linear-gradient(royalblue, royalblue)',
  backgroundPositionY: '100%',
  backgroundPositionX: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color:
    menuType === 'list'
      ? darkMode
        ? '#212121'
        : 'darkslategray'
      : !darkMode
      ? 'darkslategray'
      : 'royalblue',
  cursor: 'pointer',
  margin: menuType === 'main' && 15,
  padding: menuType === 'list' ? 20 : 8,
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
  width: '100%',
  ':hover': {
    backgroundColor: menuType === 'list' && 'lightblue',
    backgroundSize: '100% 0.1em',
    backgroundPositionX: '0%',
    color:
      menuType === 'list'
        ? darkMode
          ? '#212121'
          : 'darkslategray'
        : !darkMode
        ? 'darkslategray'
        : 'royalblue',
  },
  svg: {
    transform: 'translate(3px, -2px)',
    width: 17,
    height: 17,
  },
}));

export const StyledRouterLink = styled(RouterLink)(({ darkMode, menuType }) => ({
  backgroundImage:
    menuType === 'list'
      ? 'linear-gradient(#212121, #212121)'
      : !darkMode
      ? 'linear-gradient(darkslategray, darkslategray)'
      : 'linear-gradient(royalblue, royalblue)',
  backgroundPositionY: '100%',
  backgroundPositionX: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color:
    menuType === 'list'
      ? darkMode
        ? '#212121'
        : 'darkslategray'
      : !darkMode
      ? 'darkslategray'
      : 'royalblue',
  cursor: 'pointer',
  margin: menuType === 'main' && 15,
  padding: menuType === 'list' ? 20 : 8,
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
  width: '100%',
  svg: {
    transform: 'translate(3px, -2px)',
    width: 17,
    height: 17,
  },
  ':hover': {
    backgroundColor: menuType === 'list' && 'lightblue',
    backgroundSize: '100% 0.1em',
    backgroundPositionX: '0%',
    color:
      menuType === 'list'
        ? darkMode
          ? '#212121'
          : 'darkslategray'
        : !darkMode
        ? 'darkslategray'
        : 'royalblue',
  },
}));
