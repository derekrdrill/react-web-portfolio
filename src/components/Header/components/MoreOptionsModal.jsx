import React from 'react';
import styled from 'styled-components';
import { Box, Button, Modal, Typography } from '@mui/material/';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEye, faFilePdf } from '@fortawesome/fontawesome-free-solid';
import { faGoogleDrive } from '@fortawesome/free-brands-svg-icons';
import resume from '../../../assets/Resume_DRD.pdf';

export const MoreOptionsModal = ({ open, handleModalClose, quickViewOpen }) => (
  <Modal open={open} onClose={handleModalClose}>
    <StyledBox>
      <Typography variant='h6' component='h2'>
        Resume options
      </Typography>
      <Typography sx={{ mt: 2 }}>Select one of the following:</Typography>
      <StyledLink href={resume} download>
        <ResumeOptionButtons>
          Download a Copy
          <ResumeOptionsButtonIcon icon={faFilePdf} />
        </ResumeOptionButtons>
      </StyledLink>
      <StyledLink
        href='https://docs.google.com/document/d/1qJIlaZflsm-oBC34z5-z2aLWVnZJJZkz/edit?usp=sharing&ouid=111233393235096509511&rtpof=true&sd=true'
        target='_blank'
      >
        <ResumeOptionButtons>
          Go to Google Doc
          <ResumeOptionsButtonIcon icon={faGoogleDrive} />
        </ResumeOptionButtons>
      </StyledLink>
      <ResumeOptionButtons onClick={quickViewOpen}>
        Quick View
        <ResumeOptionsButtonIcon icon={faEye} />
      </ResumeOptionButtons>
      <ResumeOptionButtons cancelbutton='true' onClick={handleModalClose}>
        Cancel
      </ResumeOptionButtons>
    </StyledBox>
  </Modal>
);

const StyledBox = styled(Box)({
  backgroundColor: 'white',
  marginLeft: '25%',
  marginTop: '18%',
  maxWidth: '50%',
  padding: 20,
  borderRadius: 5,
});

const ResumeOptionButtons = styled(Button)(({ cancelbutton }) => ({
  backgroundColor: cancelbutton ? 'gainsboro' : '#311b92',
  color: cancelbutton ? 'black' : 'white',
  marginRight: 4,
  marginTop: 5,
  ':hover': {
    backgroundColor: cancelbutton ? 'gray' : '#6200ea',
  },
}));

const ResumeOptionsButtonIcon = styled(FontAwesomeIcon)({
  marginLeft: 5,
});

const StyledLink = styled.a({
  textDecoration: 'none',
});
