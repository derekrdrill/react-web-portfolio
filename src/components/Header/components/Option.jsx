import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Divider, Grid, MenuItem } from '@mui/material/';

export const Option = ({ darkMode, handleModalOpen, handleClose, option }) => {
  const { title, icon, href, target, divider, modal } = option;

  const menuItem = (
    <Grid className='menu-item' container spacing={3}>
      <Grid item xs={2}>
        {icon}
      </Grid>
      <Grid item xs={9}>
        {title}
      </Grid>
    </Grid>
  );

  return (
    <>
      <StyledMenuItem darkMode={darkMode} onClick={modal ? handleModalOpen : handleClose}>
        {href ? (
          <StyledMenuLink className='link' href={href} target={target}>
            {menuItem}
          </StyledMenuLink>
        ) : (
          menuItem
        )}
      </StyledMenuItem>
      {divider && <Divider />}
    </>
  );
};

Option.propTypes = {
  darkMode: PropTypes.bool,
  handleModalOpen: PropTypes.func,
  handleClose: PropTypes.func,
  option: PropTypes.object,
};

export const StyledMenuItem = styled(MenuItem)(({ darkMode }) => ({
  ':hover': {
    backgroundColor: darkMode ? 'black' : 'slategrey',
    color: 'white',
    a: {
      color: 'white',
    },
  },
}));

export const StyledMenuLink = styled.a({
  color: 'black',
  textDecoration: 'none',
  ':hover': {
    color: 'white',
  },
});
