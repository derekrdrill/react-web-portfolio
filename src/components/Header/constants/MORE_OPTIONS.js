import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEnvelopeSquare, faFile, faCode } from '@fortawesome/fontawesome-free-solid';
import { faGithubAlt, faLinkedin } from '@fortawesome/free-brands-svg-icons';

export const MORE_OPTIONS = [
  {
    id: 1,
    title: 'Email',
    icon: <FontAwesomeIcon icon={faEnvelopeSquare} />,
    href: 'mailto:derekrdrill@gmail.com',
    target: '',
  },
  {
    id: 2,
    title: 'LinkedIn',
    icon: <FontAwesomeIcon icon={faLinkedin} />,
    href: 'https://www.linkedin.com/in/derek-drill',
    target: '_blank',
  },
  {
    id: 3,
    title: 'Github',
    icon: <FontAwesomeIcon icon={faGithubAlt} />,
    href: 'https://github.com/derekrdrill',
    target: '_blank',
    divider: true,
  },
  {
    id: 4,
    title: 'Resume',
    icon: <FontAwesomeIcon icon={faFile} />,
    target: '',
    href: '',
    modal: {
      modalTitle: 'Select one',
      modalDescription: 'Would you like to download a copy or be sent to the Google doc?',
      modalButton1: 'Download .DOCX',
      modalButton2: 'Go to Google Doc',
      modalCloseButton: 'Cancel',
    },
  },
  {
    id: 5,
    title: 'Source Code',
    icon: <FontAwesomeIcon icon={faCode} />,
    target: '',
    href: '',
  },
];
