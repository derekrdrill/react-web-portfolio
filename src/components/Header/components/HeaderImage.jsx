import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { HeaderImageContainer } from './HeaderImageContainer';

import bitmojiThinking from '../../../assets/bitmoji_thinking.png';

export const HeaderImage = ({ currentRoute, homeClick }) => (
  <HeaderImageContainer condition={currentRoute === '/'}>
    <StyledHeaderImage
      src={bitmojiThinking}
      alt='home'
      onClick={currentRoute !== '/' ? homeClick : null}
    />
  </HeaderImageContainer>
);

HeaderImage.propTypes = {
  currentRoute: PropTypes.string,
  homeClick: PropTypes.func,
};

const StyledHeaderImage = styled.img({
  height: 65,
  top: 10,
  left: 10,
  cursor: 'pointer',
  ':hover': {
    opacity: 0.8,
  },
});
