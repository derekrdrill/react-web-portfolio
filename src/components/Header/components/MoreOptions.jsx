import React, { useState } from 'react';
import styled from 'styled-components';
import SettingsTwoToneIcon from '@mui/icons-material/SettingsTwoTone';
import { Menu, MenuItem, Divider, IconButton, Grid } from '@mui/material/';
import { MoreOptionsModal } from './MoreOptionsModal';
import { FileQuickView } from '../../FileQuickView/FileQuickView';
import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';
import resumePDF from '../../../assets/Resume_DRD.pdf';

const Option = props => {
  const { title, icon, href, target, divider, modal } = props.option;

  const menuItem = (
    <Grid container spacing={3}>
      <Grid item xs={2}>
        {icon}
      </Grid>
      <Grid item xs={9}>
        {title}
      </Grid>
    </Grid>
  );

  return (
    <div>
      <StyledMenuItem onClick={modal ? props.handleModalOpen : props.handleClose}>
        {href ? (
          <StyledMenuLink href={href} target={target}>
            {menuItem}
          </StyledMenuLink>
        ) : (
          menuItem
        )}
      </StyledMenuItem>
      {divider && <Divider />}
    </div>
  );
};

export const MoreOptions = MoreOptionsProps => {
  const [anchorEl, setAnchorEl] = useState(null);
  const [modalOpen, setModalOpen] = useState(false);
  const [quickViewVisibility, setQuickViewVisibility] = useState(null);
  const [loading, setLoading] = useState(false);

  const open = Boolean(anchorEl);
  const openModal = Boolean(modalOpen);
  const openQuickView = Boolean(quickViewVisibility);

  const handleClose = () => setAnchorEl(null);
  const handleClick = event => setAnchorEl(event.currentTarget);

  const handleModalOpen = () => setModalOpen(true);
  const handleModalClose = () => setModalOpen(null);

  const handleQuickViewOpen = () => {
    setLoading(true);
    setTimeout(() => {
      handleModalClose();
      setLoading(null);
      setQuickViewVisibility(true);
    }, 500);
  };

  const handleQuickViewClose = () => {
    handleModalClose();
    setQuickViewVisibility(null);
  };

  return (
    <MoreOptionsButtonContainer>
      <IconButton onClick={handleClick}>
        <StyledIcon />
      </IconButton>
      <StyledMenu anchorEl={anchorEl} open={open} onClick={handleClose}>
        {MoreOptionsProps.moreOptions.map(option => (
          <Option key={option.id} option={option} handleClose={handleClose} handleModalOpen={handleModalOpen} />
        ))}
      </StyledMenu>
      <MoreOptionsModal open={openModal} handleModalClose={handleModalClose} quickViewOpen={handleQuickViewOpen} />
      <FileQuickView
        open={openQuickView}
        onClose={handleQuickViewClose}
        fileType='pdf'
        filePath={resumePDF}
        onlineLink='https://drive.google.com/file/d/1S1MgI2aJGHlXGDBgNwIKs_78_BsnskzV/view?usp=sharing'
      />
      <LoaderSpinner open={loading} />
    </MoreOptionsButtonContainer>
  );
};

const MoreOptionsButtonContainer = styled.div({
  marginRight: 20,
  display: 'inline-block',
  ':hover': {
    opacity: 0.4,
  },
});

const StyledMenu = styled(Menu)({
  '.MuiPaper-root': {
    width: 150,
    backgroundColor: 'white',
    opacity: 0.7,
  },
});

const StyledIcon = styled(SettingsTwoToneIcon)({
  color: 'gainsboro',
});

const StyledMenuLink = styled.a({
  color: 'black',
  textDecoration: 'none',
  ':hover': {
    color: 'white',
  },
});

const StyledMenuItem = styled(MenuItem)({
  ':hover': {
    backgroundColor: 'black',
    color: 'white',
  },
});
