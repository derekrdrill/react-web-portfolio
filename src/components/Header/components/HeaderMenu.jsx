import React from 'react';
import styled from 'styled-components';
import { HeaderMenuOptions } from './HeaderMenuOptions';
import { HEADER_MENU_OPTIONS } from '../constants/HEADER_MENU_OPTIONS';

export const HeaderMenu = ({ headerType }) => (
  <HeaderMenuContainer>
    {HEADER_MENU_OPTIONS.map(OPTION => (
      <HeaderMenuOptions
        key={OPTION.id}
        menuTitle={OPTION.menuTitle}
        menuOptions={OPTION.menuOptions}
        menuIcon={OPTION.menuIcon}
        mainTo={OPTION.mainTo}
        secondaryTo={OPTION.secondaryTo}
        headerType={headerType}
      />
    ))}
  </HeaderMenuContainer>
);
export default HeaderMenu;

const HeaderMenuContainer = styled.div({
  margin: '0 auto',
});
