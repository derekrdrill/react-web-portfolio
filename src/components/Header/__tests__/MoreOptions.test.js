/*** @jest-environment jsdom */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { MoreOptions, StyledIcon, StyledMenu } from '../components/MoreOptions';

import { MORE_OPTIONS } from '../constants/MORE_OPTIONS';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('More Options tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ darkMode: true });

    renderer.render(<MoreOptions moreOptions={MORE_OPTIONS} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders StyledIcon styled component correctly', () => {
    const styledIcon = shallow(<StyledIcon />);
    const styledIconDark = shallow(<StyledIcon darkMode />);

    expect(styledIcon.props().className).toEqual('sc-fbJfA eVNmfy');
    expect(styledIconDark.props().className).toEqual('sc-fbJfA dweWQa');
  });

  it('renders StyledMenu styled component correctly', () => {
    const styledMenu = shallow(<StyledMenu />);
    const styledMenuDark = shallow(<StyledMenu darkMode />);

    expect(styledMenu.props().className).toEqual('sc-csCMJt dzYCX');
    expect(styledMenuDark.props().className).toEqual('sc-csCMJt jHULKV');
  });
});


