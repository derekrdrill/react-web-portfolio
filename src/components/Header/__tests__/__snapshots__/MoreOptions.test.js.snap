// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More Options tests renders correctly 1`] = `
<styled.div>
  <ForwardRef(IconButton)
    onClick={[Function]}
  >
    <Styled(Component) />
  </ForwardRef(IconButton)>
  <Styled(Component)
    anchorEl={null}
    onClick={[Function]}
    open={false}
  >
    <Option
      handleClose={[Function]}
      handleModalOpen={[Function]}
      option={
        Object {
          "href": "mailto:derekrdrill@gmail.com",
          "icon": <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f199",
                  "M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zM178.117 262.104C87.429 196.287 88.353 196.121 64 177.167V152c0-13.255 10.745-24 24-24h272c13.255 0 24 10.745 24 24v25.167c-24.371 18.969-23.434 19.124-114.117 84.938-10.5 7.655-31.392 26.12-45.883 25.894-14.503.218-35.367-18.227-45.883-25.895zM384 217.775V360c0 13.255-10.745 24-24 24H88c-13.255 0-24-10.745-24-24V217.775c13.958 10.794 33.329 25.236 95.303 70.214 14.162 10.341 37.975 32.145 64.694 32.01 26.887.134 51.037-22.041 64.72-32.025 61.958-44.965 81.325-59.406 95.283-70.199z",
                ],
                "iconName": "envelope-square",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />,
          "id": 1,
          "target": "",
          "title": "Email",
        }
      }
    />
    <Option
      handleClose={[Function]}
      handleModalOpen={[Function]}
      option={
        Object {
          "href": "https://www.linkedin.com/in/derek-drill",
          "icon": <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f08c",
                  "M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z",
                ],
                "iconName": "linkedin",
                "prefix": "fab",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />,
          "id": 2,
          "target": "_blank",
          "title": "LinkedIn",
        }
      }
    />
    <Option
      handleClose={[Function]}
      handleModalOpen={[Function]}
      option={
        Object {
          "divider": true,
          "href": "https://github.com/derekrdrill",
          "icon": <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  480,
                  512,
                  Array [],
                  "f113",
                  "M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z",
                ],
                "iconName": "github-alt",
                "prefix": "fab",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />,
          "id": 3,
          "target": "_blank",
          "title": "Github",
        }
      }
    />
    <Option
      handleClose={[Function]}
      handleModalOpen={[Function]}
      option={
        Object {
          "href": "",
          "icon": <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f15b",
                  "M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z",
                ],
                "iconName": "file",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />,
          "id": 4,
          "modal": Object {
            "modalButton1": "Download .DOCX",
            "modalButton2": "Go to Google Doc",
            "modalCloseButton": "Cancel",
            "modalDescription": "Would you like to download a copy or be sent to the Google doc?",
            "modalTitle": "Select one",
          },
          "target": "",
          "title": "Resume",
        }
      }
    />
    <Option
      handleClose={[Function]}
      handleModalOpen={[Function]}
      option={
        Object {
          "href": "",
          "icon": <FontAwesomeIcon
            beat={false}
            beatFade={false}
            border={false}
            bounce={false}
            className=""
            fade={false}
            fixedWidth={false}
            flip={false}
            icon={
              Object {
                "icon": Array [
                  640,
                  512,
                  Array [],
                  "f121",
                  "M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z",
                ],
                "iconName": "code",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            maskId={null}
            pull={null}
            pulse={false}
            rotation={null}
            shake={false}
            size={null}
            spin={false}
            spinPulse={false}
            spinReverse={false}
            swapOpacity={false}
            symbol={false}
            title=""
            titleId={null}
            transform={null}
          />,
          "id": 5,
          "target": "",
          "title": "Source Code",
        }
      }
    />
  </Styled(Component)>
  <MoreOptionsModal
    handleModalClose={[Function]}
    open={false}
    quickViewOpen={[Function]}
  />
  <FileQuickView
    filePath={
      Object {
        "process": [Function],
      }
    }
    fileType="pdf"
    onClose={[Function]}
    onlineLink="https://drive.google.com/file/d/1S1MgI2aJGHlXGDBgNwIKs_78_BsnskzV/view?usp=sharing"
    open={false}
  />
  <LoaderSpinner
    open={false}
  />
</styled.div>
`;
